operator/metrics/certloader.go

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package metrics
+
+import (
+	"log/slog"
+
+	"github.com/cilium/hive/cell"
+	"github.com/cilium/hive/job"
+	"github.com/spf13/pflag"
+
+	"github.com/cilium/cilium/pkg/crypto/certloader"
+	"github.com/cilium/cilium/pkg/promise"
+)
+
+const (
+	// OperatorPrometheusEnableTLS enable TLS for prometheus server
+	OperatorPrometheusEnableTLS = "operator-prometheus-enable-tls"
+
+	// OperatorPrometheusTLSCertFile specifies path to TLS certificate file
+	// for prometheus server. The file must contain PEM encoded data.
+	OperatorPrometheusTLSCertFile = "operator-prometheus-tls-cert-file"
+
+	// OperatorPrometheusTLSKeyFile specifies path to TLS private key file
+	// for prometheus server. The file must contain PEM encoded data.
+	OperatorPrometheusTLSKeyFile = "operator-prometheus-tls-key-file"
+
+	// OperatorPrometheusTLSCAFiles specifies path to one or more TLS CA certificates files
+	// to use for TLS with mutual authentication (mTLS) for prometheus server.
+	// The files must contain PEM encoded data.
+	OperatorPrometheusTLSCAFiles = "operator-prometheus-tls-ca-files"
+)
+
+type tlsConfigPromise promise.Promise[*certloader.WatchedServerConfig]
```

Comment:
```suggestion
type prometheusTLSConfigPromise promise.Promise[*certloader.WatchedServerConfig]
```
Let's specify this is for prometheus.